packages/plugin-react/src/index.ts

```diff
@@ -122,13 +116,27 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
   const viteBabel: Plugin = {
     name: 'vite:react-babel',
     enforce: 'pre',
-    config() {
+    config(_, { mode }) {
+      const isProduction =
```

Comment:
Probably worth adding a comment here, saying why this value is using NODE_ENV and VITE_USER_NODE_ENV.